### ISSUE TITLE
 Enable torch.nn.modules.padding typechecks during CI

### ISSUE THREAD

**Issue by guilhermeleobas** [pytorch/pytorch#49492]
`TestTypeHints.test_run_mypy` should pass with `[mypy-torch.nn.modules.padding]` deleted from mypy.ini

cc @ezyang @malfet @rgommers @xuzhao9 @gramster
